v1/ao/internal/reporter/methods.go

```diff
@@ -245,6 +264,10 @@ func (p *PingMethod) MessageLen() int64 {
 	return 0
 }
 
+func (p *PingMethod) Message() [][]byte {
```

Comment:
This and above Message() APIs,
since the receiver doesn't change itself, it there a benefit for using pointer? 
(don't know much about golang pointer)

Reply:
Good catch. Go's coding convention recommends to use consistent receiver types for all the methods and use pointer when in doubt, so for this reason all the receivers for this struct are pointers.

```
Next is consistency. If some of the methods of the type must have pointer receivers, the rest should too, so the method set is consistent regardless of how the type is used. See the section on method sets for details.
```
https://golang.org/doc/faq#methods_on_values_or_pointers